#172 Fix gap between actions bar and page content

BenjaminFarquhar opened the pull request:
Fix a bug where if anything was below the KeyboardActions widget, it created a gap between the actions bar and page content

diegoveloper commented:
Hey @BenjaminFarquhar , would you mind adding a new example reproducing the issue you have now? and then we could apply your changes and see if it works without breaking anything.

Create the "Custom Sample 5" in the example project.

diegoveloper commented:
The sample 3 is not working correctly after your changes:

before:

https://user-images.githubusercontent.com/4898256/147484486-1dd16509-a124-4e88-b45b-1b609f73f11e.mov

after:

https://user-images.githubusercontent.com/4898256/147484495-7df687d6-b230-4f56-94ef-a6f315ab4d46.mov

BenjaminFarquhar commented:
@diegoveloper Sample 3 does not animate to the text fields when you click a text field, even without my changes. I have checked out 1d5aa7981f2e19995b466d5463e3d8c3e8921b2c "checkout updated" and it behaves as your bottom video on custom sample 3

diegoveloper commented:
did you try on iOS?

BenjaminFarquhar commented:
@diegoveloper Yes, the truth is in the middle; I see that my changes did make the field not 100% visible when clicking the down arrow. I will fix that. However, clicking on text fields does not animate over to the clicked field before my changes.

BenjaminFarquhar commented:
@diegoveloper I have fixed the bug that was shown by using Sample 3. I have also added a Sample 5, which I have not yet committed. Do you want Sample 5 in a separate PR, or can it go in this one? Here are the videos of me using Sample 5 before and after my fix. 

Before:


https://user-images.githubusercontent.com/26356162/147502617-09f3f8b7-4b1f-4f1f-b053-747d1f223a94.mp4

After:



https://user-images.githubusercontent.com/26356162/147502745-b749a6ef-5263-4d31-9af5-0427ff41fe81.mp4

diegoveloper commented:
It's ok if you add the sample on this PR

BenjaminFarquhar commented:
@diegoveloper  Actually, wait one moment, I'm seeing the issue with Sample 3 again.

BenjaminFarquhar commented:
@diegoveloper  I'm not seeing the issue anymore after I restarted the app. 

It keeps wanting to change these files but I didn't commit them, not sure if that makes a difference. Could you test and confirm it is working on sample 3 and sample 5? Thanks.


<img width="474" alt="Screenshot 2021-12-28 at 9 09 17 AM" src="https://user-images.githubusercontent.com/26356162/147504086-aef8fda9-3faa-402a-a48a-bdcbb5cf265c.png">

BenjaminFarquhar commented:
To test without my change, simply comment out line 432.

diegoveloper commented:
ok. I'll take a look today later, thank you :)

BenjaminFarquhar commented:
It is worth noting that this bug happens occasionally even without my changes (click field 6 or 7 and for the first time, it is cutting off part of the field):

https://user-images.githubusercontent.com/26356162/147507418-b6998c51-f193-43ef-a1f4-9519f44226c8.mp4

diegoveloper reviewed (approved):
LGTM, thanks for the contribution

diegoveloper commented:
Thanks @BenjaminFarquhar , I also noticed that issue on sample 3 but it's randomly, really weird, maybe related to the delay time.

BenjaminFarquhar commented:
Sorry @diegoveloper I should have abandoned. I have noticed a bug where the `_widgetHeight` is still occasionally wrong because it is subtracting the keyboard height (which results in that weird cutting off half the textfield) and I have found a fix. I will just do another PR with that minor change

diegoveloper commented:
ok no problem, waiting for your new PR, I won't publish a new version until we have a fix for this

woelmer commented:
@BenjaminFarquhar Using version 3.4.6, I am getting a null exception on a change you made in this PR.  `_keyParent.currentContext!.findRenderObject()` currentContext is null obviously and the ! throws an exception.

BenjaminFarquhar commented:
@woelmer Ah, okay, thanks. It just needs a null check added in. I can hopefully do that this weekend sometime. 

CC: @diegoveloper

diegoveloper commented:
> @woelmer Ah, okay, thanks. It just needs a null check added in. I can hopefully do that this weekend sometime.
> 
> CC: @diegoveloper
sounds good.

BenjaminFarquhar commented:
@woelmer @diegoveloper  I just did a PR. I hope it fixes it. Just in understanding the problem, I'm not sure it was introduced in my PR. `_keyParent.currentContext!.findRenderObject()` was inside a null check where I used it. The code that I edited for this bug fix today was 2 years old, and was the only place `_keyParent.currentContext!` was used without a null check. But maybe there was some side effect from my code that caused `_keyParent.currentContext!` to become null where it hadn't previously. Nonetheless, I hope this resolves it, let me know if it doesn't.
